### DISCUSSION TITLE
Is there a support for AES256 with SHA256 (anything other than SHA1) in FIPs mode for password based encryption ??

### DISCUSSION THREAD

**Discussion by manoj4617** [openssl/openssl#24667] [Q&A]
Is there a replacement for PBE using 3DES with SHA1  **(pbeWithSHA1And3-KeyTripleDES-CBC)** in OpenSSL 3. x?  I could not find the OID for any latest algorithm such as AES256 with SHA256. If there is a replacement for it where can I get the OID and if not is there a plan to add support for more secure algorithms for PBE in future?

**Comment by kroeckx** [comment]
I think you're looking for PBKDF2

**Reply by manoj4617** [reply]
Currently we have our own implementation for PBKDF which was written long back, but since our code base relies on OIDs  of algorithms I wanted to know if OpenSSL has OID for PBE  with AES and SHA256  like it has for pbeWithSHA1And3-KeyTripleDES-CBC. Running the openssl-3 command (pic attached) shows does not show AES related algorithm. Is there any other way to get OID or is there any other way to replace DES in PBE with newer combination ???
![oid_des](https://github.com/openssl/openssl/assets/63506938/76826a45-0d1b-42fc-b178-18c835cc61be)

**Reply by kroeckx** [reply]
From RFC 8018:
```
   PBES1 combines the PBKDF1 function [Section 5.1] with an underlying
   block cipher, which shall be either DES [NIST46] or RC2 [RFC2268] in
   cipher block chaining (CBC) mode [NIST81].  PBES1 is compatible with
   the encryption scheme in PKCS #5 v1.5 [PKCS5_15].

   PBES1 is recommended only for compatibility with existing
   applications, since it supports only two underlying encryption
   schemes, each of which has a key size (56 or 64 bits) that may not be
   large enough for some applications.
```

**Reply by manoj4617** [reply]
So, correct me if I am wrong, but there is no stronger alternative to pbeWithSHA1And3-KeyTripleDES-CBC if we go with PBES2 there is no OID for the combination of algorithms for PBE ?

**Reply by kroeckx** [reply]
PBES2 does not use a single OID, if that's your question.

**Reply by manoj4617** [reply]
Ok got it 
Thank you
